sound/soc/sof/intel/mtl.c

```diff
@@ -568,6 +576,7 @@ static void mtl_ipc_dump(struct snd_sof_dev *sdev)
 
 static int mtl_dsp_disable_interrupts(struct snd_sof_dev *sdev)
 {
+	hda_sdw_int_enable(sdev, false);
```

Comment:
is this correct @plbossart . We have mtl_enable_sdw_irq() but you're not using that?

Reply:
it's the same in the end. for MTL, a call to hda_sdw_int_enable() will result in a call  to mtl_enable_sdw_irq()

````c
void hda_sdw_int_enable(struct snd_sof_dev *sdev, bool enable)
{
	const struct sof_intel_dsp_desc *chip;

	chip = get_chip_info(sdev->pdata);
	if (chip && chip->enable_sdw_irq)
		chip->enable_sdw_irq(sdev, enable);
		
mtl.c
		.enable_sdw_irq = mtl_enable_sdw_irq,
````

I preferred to use the higher-level function for consistency with other platforms.

Reply:
In this particular case I would just call `mtl_enable_sdw_irq(sdev, false)` for now, I expect to see further down the line that these steps will move to a generic code where the callback actually make sense.

With mtl code calling mtl function from mtl.c via abstraction makes things harder to follow.

Reply:
ok, if there are 2 reviewers who think it's confusing that's a clear sign I was in the wrong. Followed the suggestion and used mtl_enable_sdw_irq(sdev, false)  directly without any abstraction/indirection